api/src/backend/tasks/jobs/scan.py

```diff
@@ -237,6 +237,14 @@ def perform_prowler_scan(
 
                     status = FindingStatus[finding.status]
                     delta = _create_finding_delta(last_status, status)
+                    first_seen = datetime.now(tz=timezone.utc)
+                    # When the delta attribute is different from new it means that previously there was a finding with the same uid and delta new, so we will obtain the first_seen attribute of that first finding
+                    if delta != Finding.DeltaChoices.NEW:
+                        first_seen = (
+                            Finding.objects.filter(uid=finding_uid, delta=Finding.DeltaChoices.NEW.value)
+                            .values("first_seen")
+                            .first()["first_seen"]
+                        )
```

Comment:
Can't we move this logic inside the `_create_finding_delta` (consider renaming it if necessary) to avoid making extra queries when not needed? In that function, we retrieve the latest finding with the same `uid` to get the `delta` value. If that finding's `first_seen` is not `None`, we can just copy it from there without having to query again.